pkg/apis/eventing/v1alpha1/rabbitmq_types.go

```diff
@@ -0,0 +1,62 @@
+package v1alpha1
+
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/webhook/resourcesemantics"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:shortName={"rmqbrokerconfig"},categories=all;knative;eventing
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RabbitmqBrokerConfig is the Schema for the RabbitmqBrokerConfig API.
+// +k8s:openapi-gen=true
+type RabbitmqBrokerConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RabbitmqBrokerConfigSpec   `json:"spec,omitempty"`
+}
+
+var _ runtime.Object = (*RabbitmqBrokerConfig)(nil)
+var _ resourcesemantics.GenericCRD = (*RabbitmqBrokerConfig)(nil)
+var _ apis.Defaultable = (*RabbitmqBrokerConfig)(nil)
+var _ apis.Validatable = (*RabbitmqBrokerConfig)(nil)
+
+type RabbitmqBrokerConfigSpec struct {
+
```

Comment:
Left empty for now. Will be filled in in a follow up PR where the RabbitmqBrokerConfig can be used as a Broker.config